src/request/src/request.js

```diff
@@ -153,21 +153,22 @@ export default class Request {
       );
     }
 
-    return this.rack.execute(this.toPlainObject()).then((response) => {
-      if (isDefined(response) === false) {
-        throw new NoResponseError();
-      }
-
-      if ((response instanceof Response) === false) {
-        response = new Response({
-          statusCode: response.statusCode,
-          headers: response.headers,
-          data: response.data
-        });
-      }
-
-      return response;
-    });
+    return this.rack.execute(this)
```

Comment:
I believe we've discussed this before, but you may want to check if this change in the request is necessary.